core/controllers/base_test.py

```diff
@@ -593,3 +595,64 @@ def test_get_items(self):
                 'param2=value%20with%20%26%20%2B%20-%20/&'
                 'param3=value%20with%20.%20%%20@%20123%20=%20!%20%3C%3E')
             self.assertDictContainsSubset(params, result)
+
+
+class ControllerClassNameTest(test_utils.GenericTestBase):
+
+    def test_controller_class_names(self):
+        """This function checks that all controller class names end with
+        either 'Handler' or 'Page'.
+        """
+        for url in main.URLS:
+            if not isinstance(url, tuple):
+                handler = str(url.handler)[8:-2].rsplit('.', 1)[1]
+                class_path = str(url.handler)[8:-2].rsplit('.', 1)[0]
+                file_name = class_path.replace('.', '/') + ".py"
+                module = importlib.import_module(class_path)
+                clazz = getattr(module, handler)
+                all_base_classes = [base_class.__name__ for base_class in
+                                    (inspect.getmro(clazz))]
+                if 'BaseHandler' in all_base_classes:
+                    class_return_type = clazz.GET_HANDLER_ERROR_RETURN_TYPE
+                    # Check that the name of the class ends with 'Handler'
+                    # if it renders JSON.
+                    if class_return_type == 'json' or (
```

Comment:
Find a way to do this more declaratively so you don't have if/elif/else. Have a mapping from return types (which should be named constants) to expected name endings, for example, and use that.

In general, do your best to not repeat code.

Reply:
@seanlip Here the if/else statements are also checking if they have a `get` function or not and also the appropriate assertion message when the corresponding class names test fail

Reply:
You can still do that declaratively. Use an appropriately-defined dict, and a for loop. Don't repeat code.